### PR TITLE
PP-6077 - Refactor Charge with gateway transaction id pact test

### PR DESCRIPTION
**Pull request by nimalank7**
Description:
- This adds a gateway_transaction_id to the state as to collapse the gateway_transaction_id pact test into awaiting_capture_request pact test

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/pact/ContractTest.java [hunk 1]
```diff
@@ -253,6 +253,28 @@ public void createChargeWithDelayedCaptureTrue(Map<String, String> params) {
         setUpSingleCharge(gatewayAccountId, chargeId, chargeExternalId, ChargeStatus.CREATED, ZonedDateTime.now(), true);
     }
 
+    @State("a charge exists")
+    public void createChargeWithDelayedCaptureTrueAndAwaitingCaptureRequestStatusAndGatewayTransactionId(Map<String, String> params) {
```
Comment (stephencdaly) [inline src/test/java/uk/gov/pay/connector/pact/ContractTest.java]: What's this state being used for? Can we just add a gateway transaction id to the charge for the state below?

Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/pact/ContractTest.java]: @stephencdaly there's a `.withTransactionId(params.get("gateway_transaction_id"))` line below

Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/pact/ContractTest.java]: @nimalank7 can the method be renamed too please, e.g. `public void aChargeExists()`

Comment (nimalank7) [reply src/test/java/uk/gov/pay/connector/pact/ContractTest.java]: Yep done

**Review by oswaldquek — APPROVED** [review]

**Pull request merged**